firmware/controllers/sensors/redundant_sensor.cpp

```diff
@@ -9,33 +9,66 @@ RedundantSensor::RedundantSensor(SensorType outputType, SensorType first, Sensor
 {
 }
 
-void RedundantSensor::configure(float maxDifference, bool ignoreSecondSensor) {
+void RedundantSensor::configure(float maxDifference, bool ignoreSecondSensor, bool averageSensors, float secondMaximum) {
 	m_maxDifference = maxDifference;
 	m_ignoreSecond = ignoreSecondSensor;
+	m_averageSensors = averageSensors;
+	m_secondMaximum = secondMaximum;
 }
 
 SensorResult RedundantSensor::get() const {
-	auto result1 = Sensor::get(m_first);
+	// Sensor 1 is always full range, i.e. 0% -> 100%
+	auto sensor1 = Sensor::get(m_first);
+
+	// If sensor 1 result is invalid, return inconsistency error
+	if (!sensor1.Valid) {
+		return UnexpectedCode::Inconsistent;
+	}
 
 	// If we're set to disable redundancy, just pass thru the first sensor
 	if (m_ignoreSecond) {
-		return result1;
+		return sensor1;
 	}
 
-	auto result2 = Sensor::get(m_second);
+	// Sensor 2 may be partial, i.e. 0% -> m_secondMaximum%
+	auto sensor2 = Sensor::get(m_second);
 
-	// If either result is invalid, return invalid.
-	if (!result1.Valid || !result2.Valid) {
+	// If sensor 2 result is invalid, return inconsistency error
+	if (!sensor2.Valid) {
 		return UnexpectedCode::Inconsistent;
 	}
 
-	// If both are valid, check that they're near one another
-	float delta = absF(result1.Value - result2.Value);
-	if (delta > m_maxDifference) {
-		return UnexpectedCode::Inconsistent;
+	float sensor2Value = sensor2.Value;
+
+	// Partial redundancy; useful for some sensors: e.g. Ford and Toyota ETCS-i
+	if (m_secondMaximum < 100) {
+		// scale the second sensor value accordingly, proportioning to the first sensor
+		sensor2Value *= m_secondMaximum / 100;
+
+		// The partial redundancy threshold, slightly less than 100% to avoid issues near full-range
+		float threshold = m_secondMaximum * 0.95;
+
+		// The scaled second sensor, proportioning it to the first sensor
+		float scaledSecond = sensor2.Value * m_secondMaximum / 100;
+
+		// Check second sensor is below partial redundancy switch-over threshold
+		if (sensor2Value > threshold) {
+			// Check first sensor is at or above partial redundancy switch-over threshold
+			if (sensor1.Value > threshold - m_maxDifference) {
+				return sensor1.Value;
```

Comment:
This could instead, with an inverted `if` condition, set `sensor2Value = sensor1.Value`, and fall-through to the averaging case if for some reason that was preferred.
Similarly could force a deviant `sensor2Value` to blow the check and leverage the catch-all `return UnexpectedCode::Inconsistent;` as well.